### PR TITLE
feat: changes

### PR DESCRIPTION
**Pull request by dojo-trading**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/dojoswap: 
        
 
  
    
    
    Error: Too many errors: 94/174 on injective
    Truncated error stack:
    
        at getAllPairs (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:44:11)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:51:20
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:64:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:9
        at async Promise.all (index 0)
        at async Promise.all (index 2)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:205:3

**Comment by ninjadotgarden** [comment]
image

These are my results on localhost. What could be wrong here ?

File: projects/dojoswap/index.js [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+async function staking() {
+
+  /// POOLS
+  const response = await fetch('https://analytics.dojo.trading/dashboard/pools/tvl');
+  const data = await response.json();
+  const keys = Object.keys(data);
+  const tvl = keys.reduce((tvl, key) => tvl + data[key].tvl, 0);
+
+  /// LSD
+  const response2 = await fetch('https://analytics.dojo.trading/dashboard/ldp');
```
Comment (g1nt0ki) [inline projects/dojoswap/index.js:17]: is there some way to fetch total amount of staked INJ on chain? 
I think it is better if we create a  new folder/listing for it, called maybe "dojoswap-lsd"

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

**Comment by g1nt0ki** [comment]
> image
> 
> These are my results on localhost. How can i go about debugging this ?

hi, missed this message, you can ignore the error messages from llamatester

**Comment by llamatester** [comment]
Error while running adapter at projects/dojoswap: 
        
 
  
    
    
    Error: Too many errors: 141/177 on injective
    Truncated error stack:
    
        at getAllPairs (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:44:11)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:51:20
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:64:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:9
        at async Promise.all (index 0)
        at async Promise.all (index 2)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:205:3

**Comment by llamatester** [comment]
Error while running adapter at projects/dojoswap: 
        
 
  
    
    
    Error: Too many errors: 84/177 on injective
    Truncated error stack:
    
        at getAllPairs (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:44:11)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:51:20
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:64:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:9
        at async Promise.all (index 0)
        at async Promise.all (index 2)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:205:3

**Comment by llamatester** [comment]
Error while running adapter at projects/dojoswap: 
        
 
  
    
    
    Error: Too many errors: 81/177 on injective
    Truncated error stack:
    
        at getAllPairs (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:44:11)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/terraswap/factoryTvl.js:51:20
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:64:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:9
        at async Promise.all (index 0)
        at async Promise.all (index 2)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:205:3

**Comment by ninjadotgarden** [comment]
@g1nt0ki I have added a new folder for dojoswap-lsd and the corresponding lsd data has been moved the same.

File: projects/dojoswap/index.js (hunk 1 above)
Comment (ninjadotgarden) [reply projects/dojoswap/index.js]: @g1nt0ki Changes have been done

**Pull request closed without merging**